src/lava/magma/runtime/message_infrastructure/__init__.py

```diff
@@ -2,63 +2,108 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # See: https://spdx.org/licenses/
 
-from ctypes import CDLL, RTLD_GLOBAL
 import os
+import platform
 
 
-def load_library():
-    lib_name = 'libmessage_infrastructure.so'
-    here = os.path.abspath(__file__)
-    lib_path = os.path.join(os.path.dirname(here), lib_name)
-    if (os.path.exists(lib_path)):
-        CDLL(lib_path, mode=RTLD_GLOBAL)
-    else:
-        print("Warn: No library file")
-    extra_lib_folder = os.path.join(os.path.dirname(here), "install", "lib")
-    if (os.path.exists(extra_lib_folder)):
-        extra_libs = os.listdir(extra_lib_folder)
-        for lib in extra_libs:
-            if '.so' in lib and ('idl' not in lib):
-                lib_file = os.path.join(extra_lib_folder, lib)
-                CDLL(lib_file, mode=RTLD_GLOBAL)
-
-
-load_library()
-
-
-from lava.magma.runtime.message_infrastructure. \
-    MessageInfrastructurePywrapper import (  # noqa
-        CppMultiProcessing,
-        ProcessType,
-        Actor,
-        ActorStatus,
-        ActorCmd,
+def _get_pure_py() -> bool:
+    pure_py_env = os.getenv("LAVA_PURE_PYTHON", 0)
+    system_name = platform.system().lower()
+    if system_name != "linux":
+        return True
+    return int(pure_py_env) > 0
+
+
+PURE_PYTHON_VERSION = _get_pure_py()
+
+if PURE_PYTHON_VERSION:
+    from abc import ABC, abstractmethod
+
+
+    class Channel(ABC):
```

Comment:
While enable PypyChannel and CppChannel together, the PypyChannel will inhert the CppChannel then overwrite method need.

Reply:
added!